src/config.rs

```diff
@@ -0,0 +1,54 @@
+use std::fs::File;
+use std::path::Path;
+use std::io::{Read, Write};
+
+use toml;
+
+pub const DEFAULT_PORT: u16 = 1234;
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct Config {
+    pub server_port: Option<u16>,
+    pub server_addr: Option<String>,
+    pub admin: Option<User>,
+    pub users: Vec<User>,
+}
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct User {
+    pub name: String,
+    pub password: String,
+}
+
+fn get_content<P: AsRef<Path>>(file_path: P) -> Option<String> {
+    let file_path = file_path.as_ref();
```

Comment:
This lines seems useless.
`File::open` also takes `AsRef<Path>` as a parameter.